hub/core/chunk_engine/write.py

```diff
@@ -14,11 +15,19 @@
 from hub.util.array import normalize_and_batchify_shape
 
 
+def write_tensor_meta(key: str, storage: StorageProvider, meta: dict):
+    storage[get_meta_key(key)] = pickle.dumps(meta)
+
+
+def write_dataset_meta(storage: StorageProvider, meta: dict):
```

Comment:
this also seems out of scope

Reply:
for chunk_engine

Reply:
this would also be sweet to add in `hub/core/structure_engine/write.py`

Reply:
again, we can rename `structure_engine` but i really like this naming & think it fits very well for it's responsibilities

Reply:
Strongly disagree. What do the words say? It's an engine... for... structure? It doesn't produce or consume structure, and it barely even declares structure. "Engine" is a strange term that means different things do everyone, and I don't really get much meaning from "structure" either besides dealing with the layout of a dataset/tensor in storage, and if that's it then there has got to be a better name than `structure engine`.

Let's make a separate task out of this since the "structure" is not fully specified yet, and I don't want to be responsible for splitting and/or relocating `write_array` just yet.

Reply:
yeah it handles the transformation of interactions with a dataset. `chunk_engine` does the heavy lifting for putting data in and out of chunks, whereas there is another part of the code that translates the API interactions to write/read the tensors. `structure_engine` is a cool term but we can name it whatever, i just know that we should separate this logic from `chunk_engine`.

Reply:
for example, reading a meta file and let's say you have multiple version of meta, like if we in the future want to support original dataset structures that appear to be different.

Reply:
Yeah of course. I was mostly responding to:
> but i really like this naming & think it fits very well for it's responsibilities

Resolving for now.